presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java

```diff
@@ -0,0 +1,85 @@
+ * limitations under the License.
+ */
+package io.prestosql.plugin.oracle;
+
+import io.prestosql.plugin.jdbc.BaseJdbcClient;
+import io.prestosql.plugin.jdbc.BaseJdbcConfig;
+import io.prestosql.plugin.jdbc.ColumnMapping;
+import io.prestosql.plugin.jdbc.ConnectionFactory;
+import io.prestosql.plugin.jdbc.JdbcTypeHandle;
+import io.prestosql.plugin.jdbc.StandardColumnMappings;
+import io.prestosql.plugin.jdbc.StatsCollecting;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.type.CharType;
+import io.prestosql.spi.type.Decimals;
+import io.prestosql.spi.type.VarcharType;
+
+import javax.inject.Inject;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Optional;
+
+import static io.prestosql.spi.type.CharType.createCharType;
+import static io.prestosql.spi.type.DecimalType.createDecimalType;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.prestosql.spi.type.VarcharType.createVarcharType;
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+public class OracleClient
+        extends BaseJdbcClient
+{
+    @Inject
+    public OracleClient(BaseJdbcConfig config, @StatsCollecting ConnectionFactory connectionFactory)
+    {
+        super(config, "\"", connectionFactory);
+    }
+
+    @Override
+    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        // Here we put TABLE and SYNONYM when the table schema is another user schema
+        return connection.getMetaData()
+                .getTables(null,
+                            schemaName.orElse(null),
+                            tableName.orElse(null),
+                            new String[] {"TABLE", "SYNONYM", "VIEW"});
+    }
+
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        int columnSize = typeHandle.getColumnSize();
+        switch (typeHandle.getJdbcType()) {
+            case Types.BLOB:
+            case Types.CLOB:
+                return Optional.of(StandardColumnMappings.varbinaryColumnMapping());
+            case Types.NUMERIC:
+                return Optional.of(StandardColumnMappings.decimalColumnMapping(createDecimalType(columnSize == 0 ? 38 : columnSize, max(typeHandle.getDecimalDigits(), 0))));
+            case Types.VARCHAR:
+            case Types.NVARCHAR:
+            case Types.LONGVARCHAR:
+            case Types.LONGNVARCHAR:
+                return (columnSize > VarcharType.MAX_LENGTH || columnSize == 0)
+                    ? Optional.of(StandardColumnMappings.varcharColumnMapping(createUnboundedVarcharType()))
+                    : Optional.of(StandardColumnMappings.varcharColumnMapping(createVarcharType(columnSize)));
```

Comment:
Could you share the Oracle's DDL when we need these NUMERIC and VARCHARs' condition? While I can understand BLOB and CLOB, I couldn't find the situation.

Also, we use ternary operator only when it's really simple. It would be better to use if-else in this case.

Reply:
yes..
the diference between NVARCHAR and VARCHAR its basically encoding.. https://asktom.oracle.com/pls/apex/f%3Fp%3D100:11:::NO:RP:P11_QUESTION_ID:9462837200346048883

for LONGVARCHAR and VARCHAR the diference is that you cannot specify the length
https://docs.oracle.com/javadb/10.8.3.0/ref/rrefsqlj15147.html

this condition is the same as presto formatter but we add de condition of "columnSize == 0"
if you think that is irrelevant we omit this and use the defaults

Reply:
change to use if else

Reply:
Understood, thanks for providing the information. Even if the encoding is different, the cuurent implementation doesn't handle it, I mean the difference between OracleClient and BaseJdbcClient is only column size condition. I think the target we need to override here is only LONGVARCHAR because VARCHAR, NVARCHAR and LONGNVARCHAR won't return 0 as the column size. 

If you agree, can we remove this toPrestoType as the first commit? We can add the enhancement later. As far as I tested, NUMERIC is handled correctly without this override and other types' (BLOB, CLOB...) frequency of use isn't so high in my opinion.

Reply:
i remove VARCHAR, NVARCHAR and LONGNVARCHAR.. but i have doubts if the LONGNVARCHAR dont have the same behavior as LONGVARCHAR

NUMERIC its needs the mapping because NUMBER on oracle could be created without precision and fails.

CLOB its used to store more than 4000 chars (previous versions to 12c dont support more than 4000char on VARCHAR, and new versions supports more chars but must be activated by dba), the normal use of CLOB is used to store JSON or logs

Reply:
removed BLOB its less used..

Reply:
@ebyhr maybe CLOB should use the same mapping as LONGVARCHAR ??